lib/pure/parsecfg.nim

```diff
@@ -482,54 +494,62 @@ proc replace(s: string): string =
     inc(i)
   result = d
 
+proc writeConfigToStream*(dict: Config, stream: Stream) =
+  ## Writes the contents of the table to the specified stream
+  ## Note: Comment statement will be ignored.
+  var section, key, value, kv, segmentChar:string
```

Comment:
I didn't touch this code, just changed it to work with streams instead of files. 

Made small refactoring of this function, also found the bug :-)